assets/js/components/SettingsNotice/SettingsNotice.js

```diff
@@ -33,12 +34,27 @@ import {
 	TYPE_SUGGESTION,
 	getIconFromType,
 } from './utils';
+import { ADSENSE_GA4_TOP_EARNING_PAGES_NOTICE_DISMISSED_ITEM_KEY as DISMISSED_KEY } from '../../modules/adsense/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { Button } from 'googlesitekit-components';
+
+const { useSelect, useDispatch } = Data;
 
 export default function SettingsNotice( props ) {
-	const { children, type, Icon = getIconFromType( type ) } = props;
+	const { children, type, dismiss, Icon = getIconFromType( type ) } = props;
+
+	const { dismissItem } = useDispatch( CORE_USER );
+
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed( DISMISSED_KEY )
+	);
 
 	const Layout = children ? SettingsNoticeMultiRow : SettingsNoticeSingleRow;
 
+	if ( isDismissed ) {
+		return null;
+	}
```

Comment:
It should return `null` if the `dismiss` property is not empty and `isDismissed` is true.